packages/application-shell/src/components/user-settings-menu/user-settings-menu.js

```diff
@@ -40,28 +40,40 @@ UserAvatar.propTypes = {
   gravatarHash: PropTypes.string.isRequired,
   firstName: PropTypes.string,
   lastName: PropTypes.string,
+  email: PropTypes.string,
   // Injected
   handleMouseOver: PropTypes.func.isRequired,
   handleMouseOut: PropTypes.func.isRequired,
   isMouseOver: PropTypes.bool.isRequired,
 };
 const UserAvatarWithHoverState = withMouseOverState(UserAvatar);
 
-export default class UserSettingsMenu extends React.PureComponent {
+export class UserSettingsMenu extends React.PureComponent {
   static displayName = 'UserSettingsMenu';
 
   static propTypes = {
     firstName: PropTypes.string,
     lastName: PropTypes.string,
+    email: PropTypes.string,
     gravatarHash: PropTypes.string.isRequired,
+
+    // withRouter
+    location: PropTypes.shape({
+      pathname: PropTypes.string.isRequired,
+    }).isRequired,
   };
 
   render() {
+    const isAccountPath = Boolean(
+      matchPath(this.props.location.pathname, {
```

Comment:
I would rather if this is a prop that's passed in.

Something like "isFullMenu"

Reply:
Yeah, I thought that too but the problem is how many levels we would need to forward that prop down the tree 🤔

Reply:
@montezume If you take a look at the tree we can not pass such prop from the parents without using withRouter at some point. I think then this is a valid solution 😅

Reply:
Can't you pass it from ApplicationShell -> AppBar -> UserSettingsMenu?

Reply:
Hm, isn't it also shown during loading though? So it would be undefined then? Maybe the initial idea was better.

Reply:
Is this still an open conversation open for improvement?